components/table/src/table.spec.ts

```diff
@@ -827,3 +857,39 @@ class TestIndicatorApp {
   color: 'error' | 'warning' = 'error';
   active = true;
 }
+
+@Component({
+  selector: 'dt-test-app',
+  template: `
+    <dt-table [dataSource]="dataSource">
+      <ng-container dtColumnDef="host" dtColumnAlign="text">
+        <dt-header-cell *dtHeaderCellDef>Host</dt-header-cell>
+        <dt-cell *dtCellDef="let row">{{ row.host }}</dt-cell>
+      </ng-container>
+
+      <dt-header-row *dtHeaderRowDef="['host']"></dt-header-row>
+      <dt-row *dtRowDef="let row; columns: ['host']"></dt-row>
+      <custom-empty-state dtEmptyState></custom-empty-state>
+    </dt-table>
+  `,
+})
+export class TestCustomEmptyStateApp {
+  dataSource: object[] = [{ host: 'host-1' }];
+}
+
+@Component({
+  selector: 'custom-empty-state',
+  providers: [
+    {
+      provide: DtEmptyState,
+      useExisting: CustomEmptyState,
+    },
+  ],
+  template: `
+    <dt-empty-state-item>
+      <dt-empty-state-item-title>No host</dt-empty-state-item-title>
+      Test message
+    </dt-empty-state-item>
+  `,
+})
+export class CustomEmptyState extends DtEmptyState {}
```

Comment:
Is there a reason why you need to extend from the `DtEmptyState`? I think this is not necessary.

Reply:
[This QueryList](https://github.com/dynatrace-oss/barista/blob/0528c6610acd32b22f0b1d9a201f06a9fcc6dcf0/components/table/src/table.ts#L117) doesn't find it unless it provides a `DtEmptyState`.

Reply:
@thomaspink I guess it would probably be better to adapt the table component to query for the directive as well?

Reply:
@myieye i see your problem now. haven't checked the usage of the empty state in the table before but i now agree with you. we would either need the solution you already did, or adapt the table component to query differently as you have mentioned. both solutions are fine for me :)